File: techila/examples/R/Features/job_input_files/inputfiles_dist.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This function contains the Worker Code, which will be distributed
# and executed on the Workers. The Jobs will access their Job-specific
# input files with the name "input.txt", which is defined in the Local
# Control Code
inputfiles_dist <- function() {
  table_contents <- read.table("input.txt")
  result <- sum(table_contents)
  return(result)
}

File: techila/examples/python/Features/job_input_files/inputfiles_dist.py
# Copyright 2012-2016 Techila Technologies Ltd.

# This file contains the Worker Code, which will be distributed
# and evaluated on the Workers. The Jobs will access their Job-specific
# input files with the name 'input.txt', which is defined in the Local
# Control Code

# Import required packages
import csv

# The function that will be executed in each Job
def inputfiles_dist():

    # Read the file 'input.txt' from the temporary working directory.
    data = list(csv.reader(open('input.txt', 'r'), delimiter=' '))

    # Sum the values onthe row.
    row_int = map(int,data[0]) # Convert to integers
    sum_row_int = sum(row_int) # Sum the values on the row

    return(sum_row_int) # Return summation as the result
